lib/bundler/gem_version_promoter.rb

```diff
@@ -129,16 +129,11 @@ def sort_dep_specs(spec_groups, locked_spec)
           b_ver <=> a_ver
         when !minor? && segments_do_not_match(:minor, a_ver, b_ver)
           b_ver <=> a_ver
+        when !unlocking_gem?(gem_name) && one_version_matches(locked_version, a_ver, b_ver)
```

Comment:
personally this would read cleaner to me as `(locked_version == a_ver || locked_version == b_ver)`